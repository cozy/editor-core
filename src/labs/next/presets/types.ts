import { FeatureFlags } from '../../../plugins/feature-flags-context/types';

export interface EditorPresetProps {
  excludes?: Set<string>;
  experimental?: Array<string>;
  featureFlags?: FeatureFlags;
}
