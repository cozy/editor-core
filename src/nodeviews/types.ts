import type { Node as PMNode } from '@atlaskit/editor-prosemirror/model';

export interface ReactNodeProps {
  selected: boolean;
}

export type ProsemirrorGetPosHandler = () => number | undefined;
export type getPosHandler = getPosHandlerNode | boolean;
export type getPosHandlerNode = () => number | undefined;
export type ReactComponentProps = { [key: string]: unknown };
export type ForwardRef = (node: HTMLElement | null) => void;
export type shouldUpdate = (nextNode: PMNode) => boolean;
