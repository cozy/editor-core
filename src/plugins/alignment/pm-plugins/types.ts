export type AlignmentState = 'start' | 'end' | 'center';
export type AlignmentPluginState = {
  align: AlignmentState;
  isEnabled?: boolean;
};
