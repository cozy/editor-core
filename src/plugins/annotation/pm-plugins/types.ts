import type { DecorationSet } from '@atlaskit/editor-prosemirror/view';
import type {
  EditorState,
  SelectionBookmark,
} from '@atlaskit/editor-prosemirror/state';
import type { PortalProviderAPI } from '@atlaskit/editor-common/portal-provider';
import type { EditorAnalyticsAPI } from '@atlaskit/editor-common/analytics';
import type {
  Dispatch,
  EventDispatcher,
} from '@atlaskit/editor-common/event-dispatcher';
import type { InlineCommentAnnotationProvider, AnnotationInfo } from '../types';

export enum ACTIONS {
  UPDATE_INLINE_COMMENT_STATE,
  SET_INLINE_COMMENT_DRAFT_STATE,
  INLINE_COMMENT_UPDATE_MOUSE_STATE,
  INLINE_COMMENT_CLEAR_DIRTY_MARK,
  ADD_INLINE_COMMENT,
  INLINE_COMMENT_SET_VISIBLE,
  CLOSE_COMPONENT,
  SET_SELECTED_ANNOTATION,
}

export interface InlineCommentPluginOptions {
  dispatch: Dispatch;
  eventDispatcher: EventDispatcher;
  portalProviderAPI: PortalProviderAPI;
  provider: InlineCommentAnnotationProvider;
  editorAnalyticsAPI: EditorAnalyticsAPI | undefined;
}
export interface InlineCommentMouseData {
  isSelecting: boolean;
}

export type InlineCommentMap = { [key: string]: boolean };
export type InlineCommentAction =
  | {
      type: ACTIONS.UPDATE_INLINE_COMMENT_STATE;
      data: InlineCommentMap;
    }
  | {
      type: ACTIONS.SET_INLINE_COMMENT_DRAFT_STATE;
      data: {
        drafting: boolean;
        editorState: EditorState;
      };
    }
  | {
      type: ACTIONS.INLINE_COMMENT_UPDATE_MOUSE_STATE;
      data: {
        mouseData: InlineCommentMouseData;
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_CLEAR_DIRTY_MARK }
  | { type: ACTIONS.CLOSE_COMPONENT }
  | {
      type: ACTIONS.ADD_INLINE_COMMENT;
      data: {
        drafting: boolean;
        inlineComments: InlineCommentMap;
        editorState: EditorState;
        selectedAnnotations: AnnotationInfo[];
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_SET_VISIBLE; data: { isVisible: boolean } }
  | {
      type: ACTIONS.SET_SELECTED_ANNOTATION;
      data: { selectedAnnotations: AnnotationInfo[] };
    };

export type InlineCommentPluginState = {
  annotations: InlineCommentMap;
  selectedAnnotations: AnnotationInfo[];
  dirtyAnnotations?: boolean;
  mouseData: InlineCommentMouseData;
  draftDecorationSet?: DecorationSet;
  bookmark?: SelectionBookmark;

  // Denotes if annotations are allowed to be create on empty nodes or nodes of whitespace (Confluence spec)
  disallowOnWhitespace: boolean;

  // Allow users to hide inline comments during editing
  isVisible: boolean;
  skipSelectionHandling: boolean;
};
