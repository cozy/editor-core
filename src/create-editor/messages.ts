import { defineMessages } from 'react-intl-next';
export const editorMessages = defineMessages({
  editorAssistiveLabel: {
    id: 'fabric.editor.editorAssistiveLabel',
    defaultMessage: 'Main content area, start typing to enter text.',
    description:
      'Text that is read out by screen reader when the main editor is in focus',
  },
});
