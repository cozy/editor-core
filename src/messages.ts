// Common Translations will live here
import commonMessages, {
  linkMessages,
  linkToolbarMessages,
  statusMessages,
  dateMessages,
} from '@atlaskit/editor-common/messages';

export { linkMessages };
export { statusMessages };
export { dateMessages };
export { linkToolbarMessages };

export default commonMessages;
