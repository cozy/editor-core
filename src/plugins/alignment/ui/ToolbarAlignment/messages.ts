import { defineMessages } from 'react-intl';

export const messages = defineMessages({
  alignment: {
    id: 'fabric.editor.alignment',
    defaultMessage: 'Alignment',
    description: 'Aligns text',
  },
});
