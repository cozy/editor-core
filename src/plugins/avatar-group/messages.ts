import { defineMessages } from 'react-intl-next';

export const avatarGroupMessages = defineMessages({
  editors: {
    id: 'fabric.editor.editors',
    defaultMessage: 'Editors',
    description:
      'classifying the people that are currently editing the document',
  },
});
