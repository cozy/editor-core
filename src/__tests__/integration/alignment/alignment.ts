import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { getDocFromElement, editable, quickInsert } from '../_helpers';
import { TableCssClassName as ClassName } from '../../../plugins/table/types';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import { messages } from '../../../plugins/block-type/messages';

const alignButton = 'button[aria-label="Text alignment"]';
const alignRightButton = 'button[aria-label="Align right"]';
const headingButton = 'button[aria-label="Font style"]';
const headingh1 = 'div[role="group"] h1';

const alignRight = async (page: any) => {
  await page.waitFor(alignButton);
  await page.click(alignButton);
  await page.waitForSelector(alignRightButton);
  await page.click(alignRightButton);
};

BrowserTestCase(
  'alignment: should be able to add alignment to paragraphs',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: 'full-page',
      allowTextAlignment: true,
    });

    await page.type(editable, 'hello');
    await alignRight(page);
    expect(
      await page.$eval(editable, getDocFromElement),
    ).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'alignment: should be able to add alignment to headings',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: 'full-page',
      allowTextAlignment: true,
    });

    await page.type(editable, 'hello');
    await page.waitFor(headingButton);
    await page.click(headingButton);
    await page.waitFor(headingh1);
    await page.click(headingh1);
    await alignRight(page);
    expect(
      await page.$eval(editable, getDocFromElement),
    ).toMatchCustomDocSnapshot(testName);
  },
);

// https://product-fabric.atlassian.net/browse/ED-6802
// TODO: test failing on safari and edge
BrowserTestCase(
  'alignment: disabled when inside special nodes',
  { skip: ['safari', 'edge'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: 'full-page',
      allowTextAlignment: true,
    });

    await page.click(`[aria-label="${messages.codeblock.defaultMessage}"]`);
    await page.waitFor(alignButton);
    const isEnabled = await page.isEnabled(alignButton);
    expect(isEnabled).toBe(false);
  },
);

//  https://product-fabric.atlassian.net/browse/ED-6802
// TODO: test failing on safari and edge
BrowserTestCase(
  'alignment: disabled when editor is disabled',
  { skip: ['safari', 'edge'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: 'full-page',
      allowTextAlignment: true,
      disabled: true,
    });
    const isEnabled = await page.isEnabled(alignButton);
    expect(isEnabled).toBe(false);
  },
);

BrowserTestCase(
  'alignment: should maintain alignment when hit return',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: 'full-page',
      allowTextAlignment: true,
    });
    await alignRight(page);
    await page.click(editable);

    const firstLine = 'this is right';
    const secondLine = 'this is still right';

    await page.type(editable, firstLine);
    await page.waitUntilContainsText(editable, firstLine);
    await page.keys(['Enter']);
    await page.type(editable, secondLine);
    await page.waitUntilContainsText(editable, secondLine);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'alignment: should be able to add alignment to selected cells',
  // @see ED-10292
  { skip: ['edge', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    const CELL = 'tbody td:first-child';

    await mountEditor(page, {
      appearance: 'full-page',
      allowTextAlignment: true,
      allowTables: {
        advanced: true,
      },
    });

    await page.click(editable);

    // Insert table
    await quickInsert(page, 'Table');
    await page.waitForSelector(CELL);
    await page.click(CELL);

    // select a column
    const controlSelector = `.${ClassName.COLUMN_CONTROLS_DECORATIONS}[data-start-index="0"]`;
    await page.waitForSelector(controlSelector);
    await page.click(controlSelector);

    await alignRight(page);
    expect(
      await page.$eval(editable, getDocFromElement),
    ).toMatchCustomDocSnapshot(testName);
  },
);
