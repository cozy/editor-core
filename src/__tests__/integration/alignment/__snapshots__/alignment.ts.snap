// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alignment: should be able to add alignment to headings 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "hello",
          "type": "text",
        },
      ],
      "marks": Array [
        Object {
          "attrs": Object {
            "align": "end",
          },
          "type": "alignment",
        },
      ],
      "type": "heading",
    },
  ],
  "type": "doc",
}
`;

exports[`alignment: should be able to add alignment to paragraphs 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "hello",
          "type": "text",
        },
      ],
      "marks": Array [
        Object {
          "attrs": Object {
            "align": "end",
          },
          "type": "alignment",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`alignment: should be able to add alignment to selected cells 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "marks": Array [
                    Object {
                      "attrs": Object {
                        "align": "end",
                      },
                      "type": "alignment",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "marks": Array [
                    Object {
                      "attrs": Object {
                        "align": "end",
                      },
                      "type": "alignment",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "marks": Array [
                    Object {
                      "attrs": Object {
                        "align": "end",
                      },
                      "type": "alignment",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`alignment: should maintain alignment when hit return 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "this is right",
          "type": "text",
        },
      ],
      "marks": Array [
        Object {
          "attrs": Object {
            "align": "end",
          },
          "type": "alignment",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "text": "this is still right",
          "type": "text",
        },
      ],
      "marks": Array [
        Object {
          "attrs": Object {
            "align": "end",
          },
          "type": "alignment",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
